### PR TITLE
Fix #3 : schema object generator

### LINKED ISSUE #3: Fix schema object generator in case of depth 2 object with no property returned
Actual case :

```
{
    [...],
    "objects1": {
        "type": [
            "array",
            "null"
        ],
        "items": {
            "type": "object",
            "properties": {
                "@id": {
                    "type": "string",
                    "pattern": "^\/api\/objects1\/[\\w-;=]+$"
                },
                "@type": {
                    "type": "string",
                    "pattern": "^Object1$"
                },
                "name": {
                    "type": [
                        "string"
                    ]
                },
                "objects2": {
                    "type": [
                        "array",
                        "null"
                    ],
                    "items": {
                        "type": "object",
                        "properties": {
                            "@id": {
                                "type": "string",
                                "pattern": "^\/api\/objects2\/[\\w-;=]+$"
                            },
                            "@type": {
                                "type": "string",
                                "pattern": "^Object2$"
                            }
                        },
                        "required": [
                            "@id",
                            "@type"
                        ]
                    }
                }
            },
            "required": [
                "@id",
                "@type",
                "name"
            ]
        }
    }
}
```

`objects2` items node should return instead :
```
{
    "type": "string",
    "pattern": "^\/api\/objects2\/[\\w-;=]+$"
}
```

### PR DESCRIPTION
**Pull request by jordanaubert**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #3 
| License       | MIT

### CODE REVIEW

File: src/SchemaGenerator/ObjectSchemaGenerator.php [hunk 1]
```diff
@@ -120,7 +120,16 @@ public function generate(\ReflectionClass $reflectionClass, array $context = [])
         }
 
         $classMetadata = $this->registry->getManagerForClass($className)->getClassMetadata($className);
-        foreach ($this->propertyInfo->getProperties($className, $context) as $property) {
+        $classProperties = $this->propertyInfo->getProperties($className, $context);
+
+        if (empty($classProperties) && 1 !== $context['depth']) {
```
Comment (vincentchalamon) [inline src/SchemaGenerator/ObjectSchemaGenerator.php]: Why `1 !== $context['depth']`?

Comment (jordanaubert) [reply src/SchemaGenerator/ObjectSchemaGenerator.php]: I had a case where I needed that, i'll get you in touch when I find it again

File: src/SchemaGenerator/ObjectSchemaGenerator.php [hunk 2]
```diff
@@ -120,13 +120,22 @@ public function generate(\ReflectionClass $reflectionClass, array $context = [])
...
+        }
+
+        foreach ($classProperties as $property) {
             $mapping = $this->populator->getMapping($classMetadata, $property);
             // Prevent infinite loop & circular references
             if (!empty($mapping['targetEntity'])) {
                 $this->path[$context['depth']] = $property;
             }
-            if ($reflectionClass->getName() === ($mapping['targetEntity'] ?? null)) {
+            if (($mapping['targetEntity'] ?? null) === $reflectionClass->getName()) {
```
Comment (vincentchalamon) [inline src/SchemaGenerator/ObjectSchemaGenerator.php]: Yoda Style

File: src/Transformer/DecimalTransformer.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace ApiExtension\Transformer;
+
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
```
Comment (vincentchalamon) [inline src/Transformer/DecimalTransformer.php]: Please set your name/email

File: src/Transformer/DecimalTransformer.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ */
+final class DecimalTransformer implements TransformerInterface
+{
+    public function supports(array $mapping, $value): bool
+    {
+        return in_array($mapping['type'], [Type::DECIMAL], true);
```
Comment (vincentchalamon) [inline src/Transformer/DecimalTransformer.php]:
```php
return Type::DECIMAL === $mapping['type'];
```

File: src/Transformer/DecimalTransformer.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function toObject(array $mapping, $value): float
+    {
+        return (float) $value;
```
Comment (vincentchalamon) [inline src/Transformer/DecimalTransformer.php]: Sadly, Doctrine parse it as `string`, so we should keep it like that

File: src/Transformer/FloatTransformer.php [hunk 6]
```diff
@@ -22,19 +22,16 @@
 {
     public function supports(array $mapping, $value): bool
     {
-        return in_array($mapping['type'], [Type::FLOAT, Type::DECIMAL], true);
+        return in_array($mapping['type'], [Type::FLOAT], true);
```
Comment (vincentchalamon) [inline src/Transformer/FloatTransformer.php]:
```php
return Type::FLOAT === $mapping['type'];
```

File: src/Transformer/DecimalTransformer.php (hunk 5 above)
Comment (jordanaubert) [reply src/Transformer/DecimalTransformer.php]: `toScalar` is used in `ApiContext` so there is no need for change here :)

**Comment by vincentchalamon** [comment]
Thanks @jordanaubert

**Pull request merged**